src/Sylius/Behat/Page/Checkout/CheckoutThankYouPage.php

```diff
@@ -38,18 +36,29 @@ public function hasThankYouMessageFor($name)
     }
 
     /**
-     * @param int $timeout
-     *
-     * @throws \InvalidArgumentException
+     * {@inheritdoc}
      */
-    public function waitForResponse($timeout)
+    public function waitForResponse($timeout, array $parameters)
     {
-        $this->getDocument()->waitFor($timeout, function () {
-            return $this->isOpen();
+        $this->getDocument()->waitFor($timeout, function () use ($parameters) {
+            return $this->isOpen($parameters);
         });
     }
 
     /**
+     * {@inheritdoc}
+     */
+    protected function getUrl(array $urlParameters = [])
+    {
+        if (!isset($urlParameters['id'])) {
+            throw new \RuntimeException(sprintf('%s getUrl method require order id', self::class));
```

Comment:
Maybe use `InvalidArgumentException`? As for message `This page requires order id to be passed as parameter`?